pinot-common/src/main/java/org/apache/pinot/common/utils/DataTable.java

```diff
@@ -46,8 +52,120 @@
   String RESIZE_TIME_MS_METADATA_KEY = "resizeTimeMs";
   String EXECUTION_THREAD_CPU_TIME_NS_METADATA_KEY = "executionThreadCpuTimeNs";
 
+  /* The MetadataKeys is used in V3, where we present metadata as Map<MetadataKeys, String>
+   * ATTENTION:
+   *  - Don't add new key which has same id/name with existing keys. Duplicate id/name is not allowed.
+   *  - Don't change id/name of existing keys.
+   *  Otherwise, backward compatibility will be broken.
+   */
+  enum MetadataKeys {
+    UNKNOWN(0, "unknown"),
+    TABLE_KEY(1, "table"), // NOTE: this key is only used in PrioritySchedulerTest
+    EXCEPTION_METADATA_KEY(2, "Exception"),
+    NUM_DOCS_SCANNED_METADATA_KEY(3, "numDocsScanned"),
+    NUM_ENTRIES_SCANNED_IN_FILTER_METADATA_KEY(4, "numEntriesScannedInFilter"),
+    NUM_ENTRIES_SCANNED_POST_FILTER_METADATA_KEY(5, "numEntriesScannedPostFilter"),
+    NUM_SEGMENTS_QUERIED(6, "numSegmentsQueried"),
+    NUM_SEGMENTS_PROCESSED(7, "numSegmentsProcessed"),
```

Comment:
Why does this not have METADATA_KEY in the suffix ?

Reply:
I believe it's left over from history. In V2, some metadata key string literal has the suffix and some not, namely the following 4 does not have the suffix: 
```
  String NUM_SEGMENTS_QUERIED = "numSegmentsQueried";
  String NUM_SEGMENTS_PROCESSED = "numSegmentsProcessed";
  String NUM_SEGMENTS_MATCHED = "numSegmentsMatched";
  String NUM_CONSUMING_SEGMENTS_PROCESSED = "numConsumingSegmentsProcessed";
  String MIN_CONSUMING_FRESHNESS_TIME_MS = "minConsumingFreshnessTimeMs";
``` 
So I copied them to keep consistence. Now, I have remove the suffix for all the keys in V3. In V3, all the keys are in a 
MetadataKeys enum), so user will know they are metadata keys even if they do not has a suffix.